core/src/main/java/com/hcl/ecomm/core/servlets/DeleteCartItemServlet.java

```diff
@@ -0,0 +1,97 @@
+package com.hcl.ecomm.core.servlets;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.HttpConstants;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.json.JSONObject;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hcl.ecomm.core.services.DeleteCartItemService;
+
+/**
+ * Servlet that delete the existing cart item by hitting the Magento service.
+ */
+@Component(
+		service = Servlet.class,
+		property = { 
+				Constants.SERVICE_DESCRIPTION + "= Delete Cart Item Servlet",
+				"sling.servlet.paths=/bin/hclecomm/deleteCartItem", 
+				"sling.servlet.method=" + HttpConstants.METHOD_PUT,
+			    "sling.servlet.extensions=json" })
+public class DeleteCartItemServlet extends SlingAllMethodsServlet {
+
+	private static final long serialVersionUID = -7980975418220422865L;
+
+	private static final Logger LOG = LoggerFactory.getLogger(DeleteCartItemServlet.class);
+
+	@Reference
+	private DeleteCartItemService deleteCartItemService;
+
+    /**
+     * doPut to  delete the cart item in magento.
+     *
+     * @param request
+     *            - sling servlet request object
+     * @param response
+     *            - sling servlet response object
+     */
+	@Override
+	protected void doPut(SlingHttpServletRequest request, SlingHttpServletResponse response)
+			throws ServletException, IOException {
+		response.setContentType("application/json");
+		response.setCharacterEncoding("UTF-8");
+		JSONObject responseObject = new JSONObject();
+
+		try {
+			responseObject.put("message", "Request failed");
+			responseObject.put("status", Boolean.FALSE);
+			
+			StringBuilder buffer = new StringBuilder();
+			BufferedReader reader = request.getReader();
+			String line;
+			while ((line = reader.readLine()) != null) {
```

Comment:
@nitheshkumar-hcl : why do we need a while loop here ?

Reply:
Because here we are reading payload values from buffer string.so we are using while loop here.